charts/common/templates/deployment-canary.yaml

```diff
@@ -0,0 +1,410 @@
+{{- if .Values.canary.enabled -}}
```

Comment:
I think this assumes a single deployment in the chart no? What if we have multiple deployments. Like we have in notification-service for example?

Reply:
I might be wrong, but I think canary deployments need their own service don't they?

Reply:
Service - no, it addresses pods directly. 

Volumes - it creates regardless of a volume. `hasVolumes` is part of the original deployment file.

> So for example, if we have notification-service and notification-relay that creates a canary for both?

Since those are two different sub-charts in notification-service chart - canary could be enabled and configured independently.